internal/service/networkmanager/vpn_attachment.go

```diff
@@ -0,0 +1,323 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+
+	// "github.com/hashicorp/terraform-provider-aws/internal/flex"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceVpnAttachment() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceVpnAttachmentCreate,
+		ReadWithoutTimeout:   resourceVpnAttachmentRead,
+		UpdateWithoutTimeout: resourceVpnAttachmentUpdate,
+		DeleteWithoutTimeout: resourceVpnAttachmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"attachment_policy_rule_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"attachment_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"core_network_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"core_network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"edge_location": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"segment_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"vpn_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+		},
+	}
+}
+
+func resourceVpnAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	coreNetworkID := d.Get("core_network_id").(string)
+	vpnARN := d.Get("vpn_arn").(string)
+	input := &networkmanager.CreateSiteToSiteVpnAttachmentInput{
+		CoreNetworkId:    aws.String(coreNetworkID),
+		VpnConnectionArn: aws.String(vpnARN),
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	log.Printf("[DEBUG] Creating Network Manager VPN Attachment: %s", input)
```

Comment:
`input` is already logged if `DEBUG` logging is enabled, so including it is not needed. You could include the core network ID and VPN ARN

```suggestion
	log.Printf("[DEBUG] Creating Network Manager VPN Attachment (%s,%s)", coreNetworkID, vpnARN)
```